dracut/30ignition/ignition-generator

```diff
@@ -27,7 +27,8 @@ cmdline_bool() {
 
 add_requires() {
     local name="$1"; shift
-    local requires_dir="${UNIT_DIR}/initrd.target.requires"
+    local target="${1:-ignition-complete.target}"
+    local requires_dir="${UNIT_DIR}/${target}.requires"
```

Comment:
We're only using `add_requires` twice: once where the default arg kicks in, and once when not. ISTM like it'd be more legible to just always pass the `target` as a second parameter so it's obvious from the call site what's going on.

Reply:
i thought the same thing when reading this

Reply:
We're actually using it three times; two of them want the default target, as will any invocations added in the future.  It seems clearer to me this way, but I can change it if y'all'd like.

Reply:
either way for me. As of right now I think it would make things a little more obvious, though if we add a bunch in the future then it would start to get redundant and having a default makes sense.